controllers/onload_controller_test.go

```diff
@@ -835,6 +835,81 @@ var _ = Describe("onload controller", func() {
 				)
 			})
 
+			It("should pass the value of preload through", func() {
+				devicePlugin := appsv1.DaemonSet{}
+				devicePluginName := types.NamespacedName{
+					Name:      onload.Name + "-onload-device-plugin-ds",
+					Namespace: onload.Namespace,
+				}
+
+				onload.Spec.DevicePlugin.Preload = ptr.To(false)
+				Expect(k8sClient.Create(ctx, onload)).To(BeNil())
+
+				Eventually(func() bool {
+					err := k8sClient.Get(ctx, devicePluginName, &devicePlugin)
+					return err == nil
+				}, timeout, pollingInterval).Should(BeTrue())
+
+				Expect(devicePlugin.Spec.Template.Spec.Containers).Should(
+					ContainElement(MatchFields(IgnoreExtras, Fields{
+						"Args": ContainElement(Equal("-setPreload=false")),
+					})),
+				)
+			})
+
+			It("should pass the value of mountOnload through", func() {
+				devicePlugin := appsv1.DaemonSet{}
+				devicePluginName := types.NamespacedName{
+					Name:      onload.Name + "-onload-device-plugin-ds",
+					Namespace: onload.Namespace,
+				}
+
+				onload.Spec.DevicePlugin.MountOnload = ptr.To(false)
+				Expect(k8sClient.Create(ctx, onload)).To(BeNil())
+
+				Eventually(func() bool {
+					err := k8sClient.Get(ctx, devicePluginName, &devicePlugin)
+					return err == nil
+				}, timeout, pollingInterval).Should(BeTrue())
+
+				Expect(devicePlugin.Spec.Template.Spec.Containers).Should(
+					ContainElement(MatchFields(IgnoreExtras, Fields{
+						"Args": ContainElement(Equal("-mountOnload=false")),
+					})),
+				)
+			})
+
+			It("should pass the values of filepaths through", func() {
```

Comment:
Shall we bundle all three with `DescribeTable`? Because it's a multi-commit PR, one commit could introduce the table, and the follow-up ones could add more `Entry`.